compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -63,9 +63,7 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
   private def initSymbols(using Context) =
     if (myValueSymbols.isEmpty) {
       myValueSymbols = List(defn.Any_hashCode, defn.Any_equals)
-      myCaseSymbols = myValueSymbols ++ List(defn.Any_toString, defn.Product_canEqual,
-        defn.Product_productArity, defn.Product_productPrefix, defn.Product_productElement,
-        defn.Product_productElementName)
+      myCaseSymbols = defn.caseClassSynthesized.toList
```

Comment:
This seems to generate a list without a predefined order.

Reply:
I may have fixed it. Now it works locally. We should re-run the tests a couple of times to make sure.

Reply:
My kingdom for a `UniqueSeq`...

Reply:
Usually we fall back to https://www.scala-lang.org/api/current/scala/collection/mutable/LinkedHashSet.html

Reply:
`OrderedSet` exists under the name `SortedSet`. But like `LinkedHashSet` it is usually less efficient than a plain `HashSet`. 

An alternative would be to define a method `toSortedList` in sets. That could be very efficient for `SortedSet`s but could still be O(n log n) for other sets.

Reply:
> `OrderedSet` exists under the name `SortedSet`. But like `LinkedHashSet` it is usually less efficient than a plain `HashSet`.

Somewhere I picked up the notion that "sorted" means "has an associated sorting function", while "ordered" means "someone gave it an order", as in prepend/append/insertion-order.  I like it the distinction, but perhaps it's not universally agreed nomenclature.

> An alternative would be to define a method `toSortedList` in sets. That could be very efficient for `SortedSet`s but could still be O(n log n) for other sets.

Looks like this the most recent thread on that: https://github.com/scala/scala-library-next/issues/19